python/cugraph/cugraph/community/louvain.py

```diff
@@ -89,6 +93,14 @@ def louvain(G, max_level=None, max_iter=None, resolution=1.0, threshold=1e-7):
 
     """
 
+    isolated_vertices = list()
+    if is_nx_graph_type(type(G)):
+        num_of_vertices = len(G.nodes)
+        isolated_vertices = [v for v in range(num_of_vertices) if G.degree[v] == 0]
```

Comment:
this seems like a slow process vs putting this code later

G, isNx = ensure_cugraph_obj_for_nx(Gin)

if (isNx)
   if (G.number_of_nodes != Gin.number_of_nodes())
   then do the slow look over every vertex degree

Reply:
Then we would be keeping two copies of the graph, G and Gin

We might need to handle isolated vertices for cugraph Graph as well (line 105)

Though we hope we would add support at C layers to handle isolated vertices before 23.12

Reply:
And `ensure_cugraph_obj_for_nx` only checks type, even if we have G and Gin,

G.number_of_nodes() and Gin.number_of_nodes() would always be the same number.

Reply:
> We might need to handle isolated vertices for cugraph Graph as well (line 104), that's more expensive though.

I don't think we will handle isolated vertices this way once we start supporting them. Ideally we should be able to just create a graph by specifying isolated vertices at the graph creation and shouldn't have to rely on this trick. I think the
```
else:
    pass
```
is unnecessary here because there is no feature allowing a user to pass isolated vertices when using cuGraph only API

Reply:
@jnke2016 Handling isolated vertices at python layer is a temporary fix.
C API will be changed soon to support isolated vertices, as we already have support at C++ layer. I'm planning to make a separate PR for that. 
These are all temporary for now.